zapalki/src/console_input/BurnCommand.java

```diff
@@ -0,0 +1,124 @@
+package console_input;
+
+import resources.Cost;
+import zapalki.Match;
+import zapalki.MatchBox;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.Random;
+import java.util.stream.Collectors;
+
+public class BurnCommand implements Command{
+    private static final String NAME = "burn";
+    private final MatchBox matchBox;
+    private final Match match;
+    private final int amount;
+
+
+    public BurnCommand(Object burnTarget,int amount) {
```

Comment:
Constructor should have no logic, you should pass MatchBox and Match as separate params